V0.3.6 cannot ‘-normaliseconf’ if ‘Listen’ isn't an array

I had problems upgrading from v0.3.5 to v0.3.6, and I didn't find the error messages helpful. Yggdrasil didn't seem to like the `/etc/yggdrasil.conf` that I used with v0.3.5, so I tried to `-normaliseconf` first:

```sh
# yggdrasil below is v0.3.6
$ yggdrasil -useconf -normaliseconf < yggdrasil.conf
panic: 13 error(s) decoding:

* 'Listen[0]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[10]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[11]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[12]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[1]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[2]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[3]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[4]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[5]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[6]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[7]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[8]' expected type 'string', got unconvertible type 'uint8'
* 'Listen[9]' expected type 'string', got unconvertible type 'uint8'

goroutine 1 [running]:
main.readConfig(0xc00009e6b9, 0xc0000889e0, 0xc00009e6ba, 0x2)
	…/yggdrasil-go/cmd/yggdrasil/main.go:85 +0x459
main.main()
	…/yggdrasil-go/cmd/yggdrasil/main.go:134 +0x1e73

# yggdrasil was not being ehlpful here, was it.
```

I got `yggdrasil` up and running again by -normaliseconf'ing first with v0.3.5, then with v0.3.6.

Yggdrasil has been compiled with “go version go1.12.7 linux/amd64” (go-pie, actually) on ArchLinux.

--- a/cmd/yggdrasil/main.go
+++ b/cmd/yggdrasil/main.go
@@ -74,6 +74,12 @@ func readConfig(useconf *bool, useconffile *string, normaliseconf *bool) *config
 	if err := hjson.Unmarshal(conf, &dat); err != nil {
 		panic(err)
 	}
+	// Check for fields that have changed type recently, e.g. the Listen config
+	// option is now a []string rather than a string
+	if listen, ok := dat["Listen"].(string); ok {
+		dat["Listen"] = []string{listen}
+	}
+	// Sanitise the config
 	confJson, err := json.Marshal(dat)
 	if err != nil {
 		panic(err)

